### DISCUSSION TITLE
Notes from setting up self hosted Ente + dev environment on a Cloud VM

### DISCUSSION THREAD

**Discussion by mnvr** [ente-io/ente#3844] [General]
I recently prepared an cloud VM for testing something in the web/desktop app. I thought I'll post the steps I followed (this is all documented elsewhere too, still posting them here in case having the specific flow described helps someone).

1. Create a new Ubuntu 22 instance (Ubuntu 24 currently has showstoppers, so using the previous LTS).

2. Install Xfce4 + xorg + xrdp to prepare for [remote desktop](https://github.com/ente-io/ente/blob/main/infra/docs/remote-desktop.md) (I needed to test something in the desktop app, which is why I needed the GUI).

3. Install [NVM](https://github.com/nvm-sh/nvm), then `nvm install 20 && corepack enable`

4. Run desktop app

    ```sh
    git clone https://github.com/ente-io/ente
    cd ente/desktop 
    git submodule update --init --recursive
    yarn install
    yarn dev
    ```

5. Prepare docker to run without sudo

    ```sh
    sudo systemctl start docker
    sudo usermod -aG docker $USER 
    newgrp docker
    ```

6. Run server

    ```
    cd ../server
    docker compose up -d
    ```

7. Restart desktop app after asking it to connect to the local museum.

    ```sh
    cd ..
    echo "NEXT_PUBLIC_ENTE_ENDPOINT = http://localhost:8080" > web/apps/photos/.env.local
    cd desktop
    yarn dev
    ```

8. Create user. To view OTP

    ```sh
    docker compose logs -n 10 museum
    ```

9. Bump up storage

    ```sh
    docker compose exec postgres env PGPASSWORD=pgpass psql -U pguser -d ente_db
    > update subscriptions set storage=2199023255552 where user = <xxx>;
    ```

That's it. I also installed emacs `apt install emacs-nox`, and I had a development environment ready.

**Comment by jpg918** [comment]
>Create a new Ubuntu 22 instance (Ubuntu 24 currently has showstoppers, so using the previous LTS).

Looking for a bit of detail about this showstopper in 24 (LTS?)… does this affect the desktop app only? Or does it also affect running the server itself to host Ente Photos on 24.04 LTS?

Thanks!

**Reply by mnvr** [reply]
Desktop app only. Note that even desktop app runs fine 24, we just need to do a [workaround for the AppArmour issue in 24](https://github.com/ente-io/ente/issues/3555#issuecomment-2392703206). 

From what I vaguely remember, I'd tried these steps first with 24, but there were many small paper cuts (e.g. even during setting up RDP etc), so at some point I just gave up and went back to the old LTS which worked fine out of the box. Maybe these issues have been ironed out in 24 since then.
